File: bindings/java/src/org/raml/amf/core/domain/Payload.java
package org.raml.amf.core.domain;

import api_modeling_framework.model.domain.ParsedPayload;
import api_modeling_framework.model.domain.ParsedType;
import org.raml.amf.core.exceptions.InvalidModelException;
import org.raml.amf.utils.Clojure;

/**
 * Created by antoniogarrote on 04/05/2017.
 */

/**
 * Schema information for a Payload associated to a particular media-type
 */
public class Payload extends DomainModel {

    static {
        Clojure.require(Clojure.API_MODELING_FRAMEWORK_MODEL_DOMAIN);
    }

    public Payload(ParsedPayload rawModel) {
        super(rawModel);
    }

    public Payload(String id) {
        super(Clojure.var(
                Clojure.API_MODELING_FRAMEWORK_MODEL_DOMAIN,
                "api-modeling-framework.model.domain/map->ParsedPayload"
                ).invoke(Clojure.map())
        );
        this.setId(id);
    }

    public String getMediaType() {
        return (String) this.wrapped().media_type();
    }

    public void setMediaType(String mediaType) {
        this.rawModel = Clojure.setKw(this.rawModel, "media-type", mediaType);
    }

    /**
     * Schema information for the payload
     * @return
     */
    public Type getSchema() {
        ParsedType type =  (ParsedType) this.wrapped().schema();
        return new Type(type);
    }

    public void setSchema(Type type) {
        this.rawModel = Clojure.setKw(this.wrapped(), "schema", type.clojureModel());
    }

    private ParsedPayload wrapped() {
        return (ParsedPayload) this.rawModel;
    }
}

File: bindings/js/src/core/domain/Payload.ts
/**
 * Created by antoniogarrote on 05/05/2017.
 */

import {DomainModel} from "./DomainModel";
import {Clojure} from "../../Clojure";
import {Type} from "./Type";

/**
 * Schema information for a Payload associated to a particular media-type
 */
export class Payload  extends DomainModel {

    public getMediaType(): string | undefined {
        return Clojure.amf_domain.media_type(this.rawModel);
    }

    public setMediaType(mediaType: string | undefined) {
        this.rawModel = Clojure.amf.update(DomainModel.domain_builder, this.rawModel, Clojure.kw("media-type"), mediaType);
    }

    /**
     * Schema information for the payload
     * @return
     */
    public getSchema(): Type | undefined {
        const type = Clojure.amf_domain.schema(this.rawModel);
        if (type != null)
            return new Type(type);
    }

    public setSchema(schema: Type | undefined) {
        this.rawModel = Clojure.amf.update(DomainModel.domain_builder, this.rawModel, Clojure.kw("media-type"), schema  != null ? schema.clojureModel() : null);
    }

    public static build(id: string): Payload {
        return new Payload(Clojure.amf.build(
            DomainModel.domain_builder,
            Clojure.amf_domain.map__GT_ParsedPayload,
            id));
    }
}